### DISCUSSION TITLE
How is the line spacing around test case input handled?

### DISCUSSION THREAD

**Discussion by gsuuon** [tree-sitter/tree-sitter#3411] [Q&A] [answered]
For example, given the test:
```
==================
Return statements
==================

func x() int {
  return 1;
}

---

(source_file
  (function_definition
    (identifier)
    (parameter_list)
    (primitive_type)
    (block
      (return_statement (number)))))
```

Is the test document going to be

```
func x() int {
  return 1;
}
```
or

```

func x() int {
  return 1;
}

```
(or something else)?

I'm trying to debug an inconsistency between `tree-sitter parse` and `tree-sitter test` results. Running tree-sitter parse gave me the expected tree but tests were failing with what I thought was the same input. I managed to get parse to produce the same tree as in test if I add a newline to the start of the parse file - I realize I'd made the incorrect assumption that leading/trailing newlines in test case input were just trimmed.

What looks like is happening is that one newline is removed from the beginning, so the two '\n\n' from the beginning blank line becomes '\n' and the example becomes: `\nfunc x() int {\n  return 1;\n}`. Removing the trailing blank line padding also causes tests to fail but I can't reproduce it in `tree-sitter parse`. What happens in that situation? Currently, to get a test case to behave the same as parsing a file with a single line I need a test case that looks like:
```
==================
Test case
==================
foo

---
```
That is, no leading blank line and keep trailing blank line. Is that test case the equivalent of a single line input file?

**Comment by amaanq** [comment, marked as answer]
They are not trimmed from the beginning (except the first newline following the `=`), but I think I see what might be confusing with trailing newlines. In your last example there are two newlines following foo, and if you were to remove the empty line it'd be one newline, but I don't think it's possible to get that to zero, which could be a nice to have.
